File: ECommerce/ECommerce.Domain/Entities/ItemPedido.cs
namespace ECommerce.Domain.Entities
{
    public class ItemPedido
    {
        public int IdPedido { get; set; }
        public virtual Pedido Pedido { get; set; }
        public int IdProduto { get; set; }
        public virtual Produto Produto { get; set; }
        public decimal PrecoVenda { get; set; }
        public int IdStatus { get; set; }
        public virtual StatusPedido Status { get; set; }
    }
}

File: ECommerce/ECommerce.Presentation/ClientApp/src/app/Entitiys/ItemPedido.ts
import { Pedido } from "./Pedido";
import { Produto } from "./Produto";
import { StatusPedido } from "./StatusPedido";

export class ItemPedido{
    IdPedido: number;
    Pedido: Pedido;
    IdProduto: number;
    Produto: Produto;
    PrecoVenda: number;
    IdStatus: number;
    Status: StatusPedido;
}
